frontend/src/composables/useTokenSelection.ts

```diff
@@ -0,0 +1,57 @@
+import { computed, Ref, ref } from 'vue';
+
+import { getTokenDecimals } from '@/services/transactions/token';
+import type { EthereumProvider } from '@/services/web3-provider';
+import { useConfiguration } from '@/stores/configuration';
+import { Token } from '@/types/config';
+import { SelectorOption } from '@/types/form';
+
+export function useTokenSelection(provider: Ref<EthereumProvider | undefined>) {
+  const configuration = useConfiguration();
+
+  const TOKENS: SelectorOption[] = configuration.chains[
+    String(provider.value?.chainId.value)
+  ]?.tokens.map((token) => getTokenSelectorOption(token));
+
+  const selectedToken: Ref<SelectorOption | undefined> = ref();
+  const selectedTokenAddress = computed(() => selectedToken.value?.value);
+
+  const switchToken = (token: SelectorOption) => {
+    selectedToken.value = token;
+  };
+
+  const addTokenToProvider = async () => {
+    if (!provider.value || !selectedToken.value) {
+      throw new Error('Provider or token missing!');
+    }
+    try {
+      const decimals = await getTokenDecimals(provider.value, selectedToken.value.value);
+
+      await provider.value.addToken({
+        address: selectedToken.value.value,
+        symbol: selectedToken.value.label,
+        decimals: Number(decimals),
+      });
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const addTokenExecutable = computed(() => provider.value && selectedToken.value);
+
+  return {
+    selectedToken,
+    selectedTokenAddress,
+    TOKENS,
```

Comment:
Same upper case issue as before. Just as a hint. So depending on the output of the other thread, please apply it here too.